src/marketplace/create_listings.ts

```diff
@@ -1,196 +1,247 @@
-      ERC1155_ABI,
-      wallet
-    );
-
-    const balance = await collectionContract.balanceOf(wallet.address, tokenId);
-  
-    let isApprovedForAll = false;
-  
+  const chainConfig = findSupportedNetwork(network)
+  if (chainConfig === undefined) {
+    program.error('Unsupported network, please select a valid network')
+    return
+  }
+
+  let nodeUrl = chainConfig.rpcUrl
+
+  if (options.projectAccessKey) {
+    nodeUrl += '/' + options.projectAccessKey
+  }
+
+  if (options.verbose) {
+    console.log(`Using node URL: ${nodeUrl}`)
+  }
+
+  const provider = new ethers.JsonRpcProvider(nodeUrl)
+  const wallet = new ethers.Wallet(privateKey, provider)
+
+  // Set up the wallet client
+  const walletClient = createWalletClient({
+    account: privateKeyToAccount(privateKey),
+    chain: polygon,
+    transport: http(nodeUrl), // Replace with your Polygon node URL
+  })
+
+  let walletAddress = wallet.address as Address
+  if (!walletAddress.startsWith('0x')) {
+    walletAddress = `0x${walletAddress}`
+  }
+
+  console.log(`Using EOA Wallet: ${walletAddress}`)
+  console.log('Collection Address:', collectionAddress)
+  console.log('Currency token address:', currency)
+  console.log('Contract Type: %s, Is 1155', options.type, isERC1155)
+
+  const listingRequest = {
+    collectionAddress,
+    walletAddress,
+    tokenId,
+    currency,
+    price,
+    quantity,
+    expiry: options.expireIn,
+  }
+
+  if (isERC1155) {
     try {
-      isApprovedForAll = await collectionContract.isApprovedForAll(
-        wallet.address,
-        SEQUENCE_MARKETPLACE_V1_ADDRESS
-      );
-    } catch (error) {
-      console.dir(error, { depth: null });
-      program.error("Error checking approval.");
+      await approvalERC1155(walletClient, listingRequest)
+    } catch (e) {
+      console.dir(e, { depth: null })
+      return
     }
-  
-    if (!isApprovedForAll) {
-      const approveTxn = getErc1155ApproveAllTransaction(
-        collectionAddress,
-        SEQUENCE_MARKETPLACE_V1_ADDRESS
-      );
-  
-      try {
-        const res = await wallet.sendTransaction(approveTxn);
-
-        if (options.verbose) {
-          console.log(`Approve transaction sent: ${res.hash}`);
-        }
-  
-        const receipt = await res.wait();
-
-        if (options.verbose) {
-          console.dir(receipt, { depth: null });
-        }
-      } catch (error) {
-        console.dir(error, { depth: null });
-        program.error("Error processing transaction, please try again.");
-      }
+  } else {
+    try {
+      await approvalERC721(walletClient, listingRequest)
+    } catch (e) {
+      console.dir(e, { depth: null })
+      return
     }
-  
-    const sequenceMarketplaceInterface = new ethers.Interface(Orderbook_ABI);
-  
-    const dataCreateRequest = sequenceMarketplaceInterface.encodeFunctionData(
-      "createRequest",
+  }
+
+  const marketplaceContract = getContract({
+    address: SEQUENCE_MARKETPLACE_V1_ADDRESS,
+    abi: Orderbook_ABI,
```

Comment:
I think we should default to V2 and only use V1 if the user adds an extra argument `--marketplaceVersion 1` or smth